techniques/system/common/1.0/update.st

```diff
@@ -349,6 +349,23 @@ bundle agent update_reports
 
 }
 
+#######################################################
+# UUID file checking
+
+bundle agent check_uuid
+{
+  classes:
+    # Abort if no uuid is defined
+    "should_not_continue" expression => "wrong_uuid|no_uuid";
+    "wrong_uuid" not => strcmp("&UUID&", "${g.uuid}");
+    "no_uuid" not => fileexists("${g.uuid_file}");
+
+  reports:
+    wrong_uuid::
+      "FATAL: The UUID in ${g.uuid_file} (${g.uuid}) doesn't match the UUID in your promises (&UUID&)";
```

Comment:
Could you replace "in your promises" by "recorded for this node in your Rudder server" ?